wallet-connect/src/client/socket.rs

```diff
@@ -234,12 +245,27 @@ impl Socket {
         // a task for sending the messages to the bridge server
         let writer = tokio::spawn(async move {
             while let Some((mid, x)) = receiver.recv().await {
-                if let (Err(_), Some(id)) = (tx.send(x).await, mid) {
-                    // not to let the requester to wait forever
-                    const ERROR_MSG: &str = "\"Failed to send message to the bridge server\"";
-                    if let Some((_id, sender)) = context.0.pending_requests.remove(&id) {
-                        let _ = sender.send(serde_json::json!(ERROR_MSG));
+                let context = context.0.clone();
+                match (tx.send(x).await, mid) {
+                    (Err(_), Some(id)) => {
+                        // not to let the requester to wait forever
+                        const ERROR_MSG: &str = "\"Failed to send message to the bridge server\"";
+                        let context = context.clone();
+                        if let Some((_id, sender)) = context.pending_requests.remove(&id) {
+                            let _ = sender.send(serde_json::json!(ERROR_MSG));
+                        }
+                    }
+                    (Ok(_), Some(id)) => {
+                        // clean up pending request after timeout
+                        let _ = tokio::spawn(async move {
+                            let context = context.clone();
+                            sleep(Duration::from_millis(context.pending_requests_timeout)).await;
+                            if let Some((_id, _sender)) = context.pending_requests.remove(&id) {
+                                // TODO Reject the request?
```

Comment:
should this also send an error message, so that the requester doesn't wait forever in https://github.com/cronos-labs/play-cpp-sdk/blob/c04df79a80b21b60f683ed7c281c68d40ff6c009/wallet-connect/src/client/socket.rs#L146 ?

Reply:
or receiver will return an error if that `_sender` gets dropped?

Reply:
it could perhaps send back some error message to indicate it timed-out, so the error log message could be better

Reply:
Yes, it may wait forever, let me add message back.

Reply:
Done

Reply:
> feel a bit sick

Take care~